webob/descriptors.py

```diff
@@ -125,8 +127,21 @@ def fdel(self):
         '<Deprecated attribute %s>' % name
     )
 
+def encode_latin1(value):
+    if isinstance(value, text_type) and not PY3:
+        return value.encode('latin-1')
+    else:
+        return value
+
+def encode_location(value):
```

Comment:
For full URL's, this should parse the URL and encode the host portion using IDNA, and the rest of the URL using url encode.